lib/matplotlib/streamplot.py

```diff
@@ -473,6 +473,10 @@ def integrate(x0, y0):
     return integrate
 
 
+class OutOfBounds(Exception):
```

Comment:
might make sense to inherit from `IndexError` here and still catch `IndexError` below (so in the case where `f` does still raise it (not sure if that is possible) everything still works?

Reply:
Yep good point. I think the code shouldn't raise `IndexError` any more as we are checking bounds prior to interpolating, but I guess a bit safer to do as you suggest so I have pushed this to the PR now. Thanks!